Frends.Excel.ConvertToXML/Frends.Excel.ConvertToXML/Excel.cs

```diff
@@ -0,0 +1,203 @@
+                                    xw.WriteEndElement();
+                                }
+                            }
+
+                            if (row_element_is_writed == true)
+                            {
+                                xw.WriteEndElement();
+                            }
+                        }
+
+                        xw.WriteEndElement();
+                    }
+                }
+
+                xw.WriteEndDocument();
+                xw.Close();
+                return builder.ToString();
+            }
+        }
+    }
+
+
+    /// <summary>
+    /// A helper method.
+    /// Converts column header index to letter, as Excel does in its GUI.
+    /// </summary>
+    /// <returns>String containing correct letter combination for column.</returns>
+    public static string ColumnIndexToColumnLetter(int colIndex)
+    {
+        var div = colIndex;
+        var colLetter = string.Empty;
+        int mod;
+        while (div > 0)
+        {
+            mod = (div - 1) % 26;
+            colLetter = (char)(65 + mod) + colLetter;
+            div = ((div - mod) / 26);
+        }
+        return colLetter;
+    }
+
+    /// <summary>
+    /// A helper method. 
+    /// Converts DateTime object to the DateFormat given as options.
+    /// Return agent's date format in default.
+    /// </summary>
+    /// <param name="date"></param>
+    /// <param name="options"></param>
+    /// <returns>string containing correct date format</returns>
+    public static string ConvertDateTimes(DateTime date, Options options)
+    {
+        // Modify the date using date format var in options.
+
+        if (options.ShortDatePattern)
+        {
+            switch (options.DateFormat)
+            {
+                case DateFormats.DDMMYYYY:
+                    return date.ToString("dd/MM/yyyy", CultureInfo.InvariantCulture);
+                case DateFormats.MMDDYYYY:
+                    return date.ToString("MM/dd/yyyy", CultureInfo.InvariantCulture);
+                case DateFormats.YYYYMMDD:
+                    return date.ToString("yyyy/MM/dd", CultureInfo.InvariantCulture);
+                case DateFormats.DEFAULT:
+                    return date.ToString(CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern);
+                default:
+                    return date.ToString(CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern);
+            }
+        }
+        else
+        {
+            switch (options.DateFormat)
+            {
+                case DateFormats.DDMMYYYY:
+                    return date.ToString("dd/MM/yyyy H:mm:ss", CultureInfo.InvariantCulture);
+                case DateFormats.MMDDYYYY:
+                    return date.ToString("MM/dd/yyyy h:mm:ss tt", CultureInfo.InvariantCulture);
+                case DateFormats.YYYYMMDD:
+                    return date.ToString("yyyy/MM/dd H:mm:ss", CultureInfo.InvariantCulture);
+                case DateFormats.DEFAULT:
```

Comment:
I don't think you need to do this check, since default already returns the same result.